src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -67,111 +78,90 @@ export const WorkspaceMenu = ({ coreStart }: Props) => {
+    const workspaceListItems = panelsWorkspaceList.map((workspace) =>
+      workspaceToItem(workspace, itemType)
     );
-    workspaceListItems.push({
-      icon: <EuiIcon type="plus" />,
-      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.createWorkspace', {
-        defaultMessage: 'Create workspace',
-      }),
-      key: WORKSPACE_CREATE_APP_ID,
-      onClick: () => {
-        window.location.assign(
-          cleanWorkspaceId(
-            coreStart.application.getUrlForApp(WORKSPACE_CREATE_APP_ID, {
-              absolute: false,
-            })
-          )
-        );
-      },
-    });
-    workspaceListItems.push({
-      icon: <EuiIcon type="folderClosed" />,
-      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.allWorkspace', {
-        defaultMessage: 'All workspaces',
-      }),
-      key: WORKSPACE_LIST_APP_ID,
-      onClick: () => {
-        window.location.assign(
-          cleanWorkspaceId(
-            coreStart.application.getUrlForApp(WORKSPACE_LIST_APP_ID, {
-              absolute: false,
-            })
-          )
-        );
-      },
-    });
     return workspaceListItems;
   };
 
-  const currentWorkspaceButton = (
-    <>
-      <EuiListGroup style={{ width: 318 }} maxWidth={false}>
-        <EuiListGroupItem
-          iconType="spacesApp"
-          label={currentWorkspaceName}
-          onClick={openPopover}
-          extraAction={{
-            color: 'subdued',
-            onClick: openPopover,
-            iconType: isPopoverOpen ? 'arrowDown' : 'arrowRight',
-            iconSize: 's',
-            'aria-label': 'Show workspace dropdown selector',
-            alwaysShow: true,
-          }}
-        />
-      </EuiListGroup>
-    </>
+  const currentWorkspaceButton = currentWorkspace ? (
+    <EuiButtonEmpty flush="left" onClick={openPopover} data-test-subj="current-workspace-button">
+      <EuiAvatar
+        size="s"
+        type="space"
+        name={currentWorkspace.name}
+        color={currentWorkspace.color}
+        initialsLength={2}
+      />
+    </EuiButtonEmpty>
+  ) : (
+    <EuiButtonIcon
+      iconType="spacesApp"
+      onClick={openPopover}
+      aria-label="workspace-select-button"
+      data-test-subj="workspace-select-button"
+    />
   );
 
-  const currentWorkspaceTitle = (
-    <EuiFlexGroup alignItems="center">
-      <EuiFlexItem grow={true}>
-        <EuiText size="s">{currentWorkspaceName}</EuiText>
-      </EuiFlexItem>
-      <EuiFlexItem grow={false}>
-        <EuiButtonIcon
-          iconType="cross"
-          onClick={closePopover}
-          aria-label="close workspace dropdown"
-        />
-      </EuiFlexItem>
-    </EuiFlexGroup>
-  );
+  const allWorkspacesPanels = [
+    {
+      id: 0,
+      title: (
+        <span className="custom-title">
+          <FormattedMessage
+            id="core.ui.primaryNav.contextMenuTitle.allWorkspaces"
```

Comment:
same here, should under workspace namespace

Reply:
updated